tools/autograd/templates/python_variable_methods.cpp

```diff
@@ -418,51 +418,131 @@ static PyObject * THPVariable_cuda(PyObject* self, PyObject* args, PyObject* kwa
   END_HANDLE_TH_ERRORS
 }
 
-static PyObject * THPVariable_to_type(PyObject* self, ScalarType scalarType) {
+static PyObject * THPVariable_to_type(PyObject* self, ScalarType scalarType, c10::optional<c10::MemoryFormat> optional_memory_format) {
   HANDLE_TH_ERRORS
   auto& self_ = reinterpret_cast<THPVariable*>(self)->cdata;
-  // Setting to MemoryFormat::Contiguous now, will change to accept memory_format in next PR
-  return THPVariable_Wrap(dispatch_to(self_, scalarType, false, false, MemoryFormat::Contiguous));
+  return THPVariable_Wrap(dispatch_to(self_, scalarType, false, false, optional_memory_format));
   END_HANDLE_TH_ERRORS
 }
-static PyObject * THPVariable_byte(PyObject* self, PyObject* args) {
-  return THPVariable_to_type(self, ScalarType::Byte);
+
+static PyObject * THPVariable_to_type_byte(PyObject* self, PyObject* args, PyObject* kwargs)  {
+  HANDLE_TH_ERRORS
+  static PythonArgParser parser({
+    "byte(*, MemoryFormat? memory_format=None)"
+  });
+  ParsedArgs<1> parsed_args;
+  auto r = parser.parse(args, kwargs, parsed_args);
+  auto opt_memory_format = r.memoryformatOptional(0);
+  return THPVariable_to_type(self, ScalarType::Byte, opt_memory_format);
+  END_HANDLE_TH_ERRORS
 }
 
-static PyObject * THPVariable_char(PyObject* self, PyObject* args) {
-  return THPVariable_to_type(self, ScalarType::Char);
+static PyObject * THPVariable_to_type_char(PyObject* self, PyObject* args, PyObject* kwargs)  {
```

Comment:
We should keep the name as `THPVariable_char`. Most, if not all autogenerated functions follow a naming format of `THPVariable_{api_name}`; I also rely on this to search through the code base (but I'm not sure if anyone else does this).